CorsixTH/Lua/world.lua

```diff
@@ -511,7 +541,12 @@ function World:spawnPatient(hospital)
   end
 
   -- Construct patient.
-  local spawn_point = self.spawn_points[math.random(1, #self.spawn_points)]
+  local desks = hospital:getStaffedDesks()
+  local spawns = self.spawn_points
+  -- Let the first patient to the player use the shortest path to a reception desk
+  local spawn_point = hospital:isPlayerHospital() and not hospital:hadPatients() and
+      self:_findClosestSpawnToDesk(desks, spawns) or spawns[math.random(1, #spawns)]
```

Comment:
Because you wrote this `and-or` construct, you always need `desks`, even if the first patient was already dispatched.

If you change it to recognize that specific case, and only then ask for desks, the hospital doesn't need a staff member to do all the paperwork about staffed desks to the world, for each patient :)

Reply:
True, maybe I shouldn't have made it a local and instead just use the function call directly.

Reply:
That would work too indeed.